pkg/api/endpoints/util.go

```diff
@@ -83,21 +65,63 @@ func RepackSubsets(subsets []api.EndpointSubset) []api.EndpointSubset {
 	// Next, build the N-to-M association the API wants.
 	final := []api.EndpointSubset{}
 	for key, ports := range addrSetsToPorts {
-		addrs := []api.EndpointAddress{}
-		for k := range addrSets[key] {
-			addrs = append(addrs, *k)
+		var addrs, notReadyAddrs []api.EndpointAddress
+		for k, ready := range addrSets[key] {
```

Comment:
can you rename k to address
all the single-letter variables in this function and the joinAddress function need to be renamed.
it is very difficult to follow what is going on...